### ISSUE TITLE
initializationFailTimeout doesn't appear to provide tolerance to unavailable DB

### ISSUE THREAD

**Issue by mattrussell-sonocent** [brettwooldridge/HikariCP#1641]
I'm struggling to get `initializationFailTimeout` to work as I might expect, although perhaps my understanding of the feature isn't correct. The context is that I'm bringing up a service and Postgres roughly at the same time in Docker Compose, and would like a bit of tolerance to Postgres being slightly slower to start up. I've set `initializationFailTimeout` to 30000 millis, but it seems that if Postgres is not available the very moment Hikari starts, I get a `com.zaxxer.hikari.pool.HikariPool$PoolInitializationException: Failed to initialize pool: The connection attempt failed.` exception (`Caused by: java.net.SocketTimeoutException: connect timed out`). Using Hikari version 3.4.5.

**Comment by TomAtPowercor** [comment]
Set it to -1. This will tell Hikari not to create any connections on startup.

**Comment by mattrussell-sonocent** [comment]
> Set it to -1. This will tell Hikari not to create any connections on startup.

Thanks, I will give that a go, that might address the use case I'm after - but does `initializationFailTimeout` just not work for positive values?

**Comment by TomAtPowercor** [comment]
Sorry, I can't answer that question as I've never tried it myself. I was just passing by looking for something else and saw this issue.

**Comment by turkogluc** [comment]
I have the same problem and setting it to -1 also does not change the behavior, so it fails immediately at the start up. In fact, what I need is having some timeout period rather than fast fail. Is there any update about this issue?

**Comment by brettwooldridge** [comment]
That seems dubious.  This is covered by several unit tests, and I just reviewed the code flows.  It actually seems like it *does* reach 30 seconds (30000ms before failing).  If I am wrong, please open a new issue with logs attached that have timestamped log messages and debug logging enabled.

**Comment by turkogluc** [comment]
I had a similar issue in a Spring project. I was setting `initializationFailTimeout` to 90 seconds in the `application.yml` file but it was failing immediately.
After I set the logging level to Debug

```
logging.level.com.zaxxer.hikari.pool.HikariPool=DEBUG
logging.level.com.zaxxer.hikari.HikariConfig=DEBUG
```

And I realized that `initializationFailTimeout` value was in fact set to `1`. So it was basically my bad to not realize that there is `DataSource` bean setting the configuration and the config from `application.yml` was not effective. That why the default value was being used, which is 1 and makes it fail immediately.

So my point is if you have checked that the value is set correctly?
